app/controllers/books_controller.rb

```diff
@@ -54,4 +63,49 @@ def book_params
          :description, :isbn, :page_count, :image_link, :language, :is_pdf,
          :quantity, :price)
     end
+
+    def publisher_params
+      params.require(:publisher).permit(:name)
+    end
+
+    def author_params
+      params.permit(authors:[:name])
+    end
+
+    def category_params
+      params.permit(categories:[:name])
+    end
+
+    def authors_list
+      author_params[:authors] || []
+    end
+
+    def categories_list
+      category_params[:categories] || []
+    end
+
+    def create_publisher
+      @publisher = Publisher.find_or_initialize_by(publisher_params)
+      if @publisher.save
```

Comment:
I think since you use find_or_initialize your condition should be like
```publisher.presisted? && publisher.save```

Reply:
okey but why?

Reply:
what are you trying to save if it is already there?

Reply:
`initialize` likes `new` doesn't save in DB if I will use `persisted?` it would give me `false`